invoke/program.py

```diff
@@ -606,10 +618,10 @@ def called_as(self):
 
         .. versionadded:: 1.2
         """
-        return os.path.basename(self.argv[0])
+        return os.path.basename(self.argv[0]) if self.argv else "invoke"
```

Comment:
This looks a lot like changing behavior! And I don't see a counterpart change elsewhere that would explain it, unless I missed one. What's up?

Reply:
I should have commented on that one instead - slops on my part. The argv is `Optional` and it should probably return `None` instead. I'll add a comment but I think we probably need to dig a little on what should happen when it is None.

Reply:
Actually, I think I modified argv earlier to default to an empty list. Accessing this out of order will throw an index error instead. What would you like to see here if the list is empty?

Reply:
@bitprophet is the empty sting fine here since it essentially means that function was provided no arguments?